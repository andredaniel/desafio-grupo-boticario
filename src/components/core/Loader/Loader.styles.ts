import styled from 'styled-components'
import { theme } from '../../../theme'

export const LoaderWrapper = styled.div`
  background-color: ${theme.colors.primary.MAIN};
  height: 100%;
  position: fixed;
  width: 100%;
  z-index: 9999;

  @keyframes rotating {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }

  &:after {
    animation: rotating 20s linear infinite;
    border-radius: 50%;
    border: 4px dotted white;
    content: '';
    height: 100px;
    left: calc(50% - 50px);
    position: absolute;
    top: calc(50% - 50px);
    width: 100px;
  }
`
