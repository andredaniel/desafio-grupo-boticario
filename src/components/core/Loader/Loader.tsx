import React from 'react'
import { LoaderWrapper } from './Loader.styles'

const Loader: React.FC = (): JSX.Element => {
  return <LoaderWrapper />
}

export { Loader }
