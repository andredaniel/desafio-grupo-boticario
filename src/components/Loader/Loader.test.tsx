import { render } from '@testing-library/react'
import React from 'react'
import { Loader } from './Loader'

describe('<Loader />', () => {
  it('should render Loader component as expected', () => {
    const { container } = render(<Loader />)
    expect(container).toMatchSnapshot()
  })
})
