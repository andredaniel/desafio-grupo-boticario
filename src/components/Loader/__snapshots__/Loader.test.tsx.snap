// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> should render Loader component as expected 1`] = `
<div>
  <div
    class="sc-bdfBwQ gSHISJ"
  />
</div>
`;
